apps/cloud_inference/server_app/source/inference_hip_rocal.cpp

```diff
@@ -0,0 +1,1092 @@
+                    if(status != VX_SUCCESS) {
+                        fatal("workDeviceProcess: vxMapTensorPatch(input)) failed(%d)", status);
+                    }
+                    DecodeScaleAndConvertToTensor(dimInput[0], dimInput[1], size, (unsigned char *)byteStream, ptr, useFp16);
+                    status = vxUnmapTensorPatch(openvx_input, map_id);
+                    if(status != VX_SUCCESS) {
+                        fatal("workDeviceProcess: vxUnmapTensorPatch(input)) failed(%d)", status);
+                    }
+                    // process the graph
+                    status = vxProcessGraph(openvx_graph);
+                    if(status != VX_SUCCESS) {
+                        fatal("workDeviceProcess: vxProcessGraph()) failed(%d)", status);
+                    }
+                    ptr = nullptr;
+                    status = vxMapTensorPatch(openvx_output, 4, NULL, NULL, &map_id, stride, (void **)&ptr, VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+                    if(status != VX_SUCCESS) {
+                        fatal("workDeviceProcess: vxMapTensorPatch(output)) failed(%d)", status);
+                    }
+                    int label = 0;
+                    float max_prob = ptr[0];
+                    for(int c = 1; c < dimOutput[2]; c++) {
+                        float prob = ptr[c];
+                        if(prob > max_prob) {
+                            label = c;
+                            max_prob = prob;
+                        }
+                    }
+                    status = vxUnmapTensorPatch(openvx_output, map_id);
+                    if(status != VX_SUCCESS) {
+                        fatal("workDeviceProcess: vxUnmapTensorPatch(output)) failed(%d)", status);
+                    }
+                    // release byteStream and keep the results in outputQ
+                    delete[] byteStream;
+                    outputQ.enqueue(std::tuple<int,int>(tag,label));
+#endif
+#elif INFERENCE_SCHEDULER_MODE == LIBRE_INFERENCE_SCHEDULER
+                    // submit the input (tag,byteStream,size) to scheduler
+                    inputQ.enqueue(std::tuple<int,char*,int>(tag,byteStream,size));
+#endif
+                }
+            }
+        }
+
+        // if nothing done, wait for sometime
+        if(!didSomething && INFERENCE_SERVICE_IDLE_TIME > 0) {
+            std::this_thread::sleep_for(std::chrono::milliseconds(INFERENCE_SERVICE_IDLE_TIME));
+        }
+    }
+    info("runInference: terminated for %s", clientName.c_str());
+
+    // send and wait for INFCOM_CMD_DONE message
+    InfComCommand reply = {
+        INFCOM_MAGIC, INFCOM_CMD_DONE, { 0 }, { 0 }
+    };
+    ERRCHK(sendCommand(sock, reply, clientName));
+    ERRCHK(recvCommand(sock, reply, clientName, INFCOM_CMD_DONE));
+
+    return 0;
+}
+
+#if INFERENCE_SCHEDULER_MODE == LIBRE_INFERENCE_SCHEDULER
+void InferenceEngineRocalHip::workMasterInputQ()
```

Comment:
if this function implementation is same as base class, you don't need to define it here

Reply:
It is not same as base class.
The rocAL version uses queueDeviceNameQ instead of the queueDeviceImageQ unlike the original class.